accord-core/src/test/java/accord/impl/basic/RandomDelayQueue.java

```diff
@@ -63,6 +64,22 @@ public int compareTo(Item that)
             return c;
         }
 
+        @Override
+        public boolean equals(Object o)
+        {
+            if (o instanceof Pending) return item.equals(o);
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+            Item item1 = (Item) o;
+            return time == item1.time && seq == item1.seq && item.equals(item1.item);
+        }
+
+        @Override
+        public int hashCode()
+        {
+            throw new UnsupportedOperationException();
```

Comment:
This debate over whether to override hash code and equals together as is strongly suggested in effective java, ending up with it throwing unsupported operation exception is the worst of both worlds.

We don't get working hash code, and we also have to scroll past the code (the reason I hear we don't want to implement hash code along with equals).

I personally have been bitten multiple times writing tests and setting expected result sets (not order sensitive) and then found out that something overrides equals, but not hash code. So yeah I want at least unsupported operation exception so I don't have to spend time debugging why I get this wrong answer.

But for the same number of lines we could have gotten the real hash code.

Reply:
The issue is not the code (though this also matters) but the *implication* that the item is used in a hash collection.

The presence of a hashCode method *should* imply to a future author that the object *is* used in a hash collection of some kind, who can may bind their decision making about how the object is evolved.

This is why we have a general policy of deleting unused methods wherever possible - the reader should know that a method’s presence has a specific purpose in the application, and draw the relevant inferences.

It sounds like you have been mislead at times by assuming that something which overrides equals also overrides hashCode, but it’s not clear why you would assume this?

Reply:
Overriding equals breaks `hashCode`. It takes a working pair of methods and violates their contract. The expectation is always that `equals` and `hashCode` will behave correctly. It's in the documentation for the method in the JRE https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#hashCode()

That's why it is item 11 in effective Java https://ahdak.github.io/blog/effective-java-part-2/#item-11--always-override-hashcode-when-you-override-equals

Reply:
> It sounds like you have been mislead at times by assuming that something which overrides equals also overrides hashCode, but it’s not clear why you would assume this?

I have personally fixed several bugs in C* where equals or hashCode is assumed to work and has not... you see one and assume the other as that is the strong stance in java; even fun ones at our `AbstractType` level (that was fun to track down)...

I strongly disagree with our current style guide as I find it causes more bugs than any perceived gains.  I am cool not defining either if not needed, but when you need one we should have the other.

Now, do I want to debate that in this PR?  No.  It's the current style guide so I would prefer to move such a convo to a dev thread.

Reply:
I’m aware of Joshua Bloch’s stance on this, and I disagree with it. But, implementing UOE satisfies both criteria, so I don’t really see the issue.